src/shared/mount-util.c

```diff
@@ -808,18 +808,11 @@ static int mount_in_namespace(
         if (r < 0)
                 return log_debug_errno(r, "Failed to retrieve FDs of the target process' namespace: %m");
 
-        if (fstat(mntns_fd, &st) < 0)
-                return log_debug_errno(errno, "Failed to fstat mount namespace FD of target process: %m");
-
-        r = namespace_open(0, NULL, &self_mntns_fd, NULL, NULL, NULL);
+        r = in_same_namespace(target, 0, NAMESPACE_MOUNT);
         if (r < 0)
-                return log_debug_errno(r, "Failed to retrieve FDs of systemd's namespace: %m");
-
-        if (fstat(self_mntns_fd, &self_mntns_st) < 0)
-                return log_debug_errno(errno, "Failed to fstat mount namespace FD of systemd: %m");
-
+                return log_debug_errno(errno, "Failed to determine if mount namespaces are equal: %m");
         /* We can't add new mounts at runtime if the process wasn't started in a namespace */
-        if (stat_inode_same(&st, &self_mntns_st))
+        if (r == 1)
```

Comment:
Please use `if (r > 0)` even if it returns 0 or 1 on success.

Reply:
Sure, can do but why? If we start returning `2` or `3` to indicate something else then we'd be misleading caller's.

Reply:
If you want to implement a function that has multiple success cases, then define an enum and return it, rather than returning 2, 3, ...
See e.g. `detect_vm()` in src/basic/virt.c.

BTW, the newly introduced function here is a boolean function with negative errno. In that case, the implicit cast to boolean from integer is also accepted if negative errno case is already handled:
```c
r = func();
if (r < 0)
        return r;
if (r)
        ...;
```
Personally, I prefer `if (r > 0)` style, but feel free to use the way you'd like.